src/services/jsTyping.ts

```diff
@@ -5,49 +5,190 @@
+                let packageJsonDict = JSON.parse(_host.readFile(packageJsonPath));
+                if (packageJsonDict.hasOwnProperty("dependencies")) {
+                    mergeTypings(Object.keys(packageJsonDict.dependencies));
+                }
+            }
+            catch(e) {
+            }
+        }
+    }
+
+    function getTypingNamesFromBowerJson(bowerJsonPath: string) {
+        if (_host.fileExists(bowerJsonPath)) {
+            // Guide against malformed package.json
+            try{
+                let packageJsonDict = JSON.parse(_host.readFile(bowerJsonPath));
+                if (packageJsonDict.hasOwnProperty("dependencies")) {
+                    mergeTypings(Object.keys(packageJsonDict.dependencies));
+                }
+            }
+            catch(e) {
+            }
+        }
+    }
+
+    /**
+     * Infer typing names from given file names. For example, the file name "jquery-min.2.3.4.js"
+     * should be inferred to the 'jquery' typing name; and "angular-route.1.2.3.js" should be inferred
+     * to the 'angular-route' typing name.
+     * @param fileNames are the names for source files in the project
+     * @param safeList is the list of names that we are confident they are library names that requires typing
+     */
+    function getTypingNamesFromSourceFileNames(fileNames: string[], safeList: string[]) {
+        safeList = safeList.map(s => s.toLowerCase());
+        fileNames = fileNames.map(f => f.toLowerCase());
+        let exactlyMatched: string[] = []
+        let notExactlyMatched: string[] = []
+        for (let fileName of fileNames) {
+            let baseName = ts.getBaseFileName(fileName);
+            let baseNameWithoutExtension = baseName.substring(0, baseName.lastIndexOf("."));
+            (safeList.indexOf(baseNameWithoutExtension) >= 0) ? exactlyMatched.push(baseNameWithoutExtension) : notExactlyMatched.push(baseNameWithoutExtension);
+        }
+        
+        let regex = /((?:\.|-)min(?=\.|$))|((?:-|\.)\d+)/g;
+        notExactlyMatched = notExactlyMatched.map(f => f.replace(regex, ""));
+        let typingNames = exactlyMatched.concat(notExactlyMatched.filter(f => ts.contains(safeList, f)));
+        mergeTypings(typingNames);
+    }
+
+    /**
+     * Infer typing names from node_module folder
+     * @param nodeModulesPath is the path to the "node_modules" folder
+     */
+    function getTypingNamesFromNodeModuleFolder(nodeModulesPath: string) {
+        // Todo: add support for ModuleResolutionHost too
+        if (!_host.fileExists(nodeModulesPath)) {
+            return;
+        }
+        
+        let typingNames: string[] = [];
+        let packageJsonFiles = 
+            _host.readDirectory(nodeModulesPath, /*extension*/undefined, /*exclude*/undefined, /*depth*/ 2).filter(f => ts.getBaseFileName(f) === "package.json");
+        for (let packageJsonFile of packageJsonFiles) {
+            let packageJsonContent = JSON.parse(_host.readFile(packageJsonFile));

```

Comment:
Try-catch when parsing json files.  Actually, how about create a TryParseJson()  that can be applied to all places instead of wrapping all chunks of code with the try catch block?  It feels like we should only be catching the ParseJson and not masking other errors.